xdsl/passes.py

```diff
@@ -116,32 +116,39 @@ def required_fields(cls: type[ModulePassT]) -> set[str]:
             field.name for field in dataclasses.fields(cls) if not _is_optional(field)
         }
 
-    def pipeline_pass_spec(self) -> PipelinePassSpec:
+    def pipeline_pass_spec(self, *, include_default: bool = False) -> PipelinePassSpec:
         """
         This function takes a ModulePass and returns a PipelinePassSpec.
+
+        If `include_default` is `True`, then add arguments that are optional are not
+        included in the spec.
```

Comment:
This sentence is kinda hard for me to parse. I think I understand it, but how about this?

```suggestion
        If `include_default` is `True`, then optional arguments are not included in the spec.
```